Tests/test_Restriction.py

```diff
@@ -38,13 +53,12 @@ def test_non_iupac_letters(self):
 
     def test_formatted_seq(self):
         """Test several methods of FormattedSeq."""
-        self.assertEqual(str(FormattedSeq(Seq("GATC"))),
-                         "FormattedSeq(Seq('GATC'), linear=True)")
-        self.assertFalse(FormattedSeq(Seq("GATC")) ==
-                         FormattedSeq(Seq("TAGC")))
+        self.assertEqual(
+            str(FormattedSeq(Seq("GATC"))), "FormattedSeq(Seq('GATC'), linear=True)"
+        )
+        self.assertFalse(FormattedSeq(Seq("GATC")) == FormattedSeq(Seq("TAGC")))
```

Comment:
@svalqui Sorry for this request, although not black-related but still style-related: Can you change all `assertFalse(a==b)` and `assertTrue(a==b)` into `assertNotEqual(a, b)` and `assertEqual(a, b)`?

Reply:
Not only here but in the whole file?

Reply:
Logged #2835 for the wider issue, but happy for you to fix this file as part of this pull request, or leave it for later.

Reply:
updated, but left one that doesn't pass the test.

```
    self.assertNotEqual(Acc65I, Asp718I)
AssertionError: Acc65I == Asp718I

```
I think is mean to use assertNotEqual.

Reply:
This is a special thing within `Bio.Restriction`, `__eq__` (`==`) and `__ne__` (`!=`) have been overriden with different meanings. In this case Acc65I and Asp718I are neither equal (they are not the same enzyme) nor not equal (they are isoschizomers, doing exactly the same thing). 
Thus `assertFalse` is the only method which allows to test both. OK.

Reply:
I'd forgotten about that quirk. Best or at least standard practise is ``__eq__`` and ``__ne__`` give the negation of each other, but as here that isn't required.

Reply:
@peterjc @MarkusPiotrowski Perhaps I'm new here, but is there any reason why the `__ne__` operator is used to check for isoschizomer-ness? It doesn't seem very convenient if it confuses the linter and tests (and potentially humans too). When I first saw this behavior in the tests, I thought it was a bug, but then realized it was by design.

Reply:
I would agree in hindsight this was a poor design choice. We have been looking at replacing the entirety of the old restriction digest module anyway.

Reply:
@peterjc Thank you for the clarification, and I look forward to seeing the new design!